code/modules/surgery/organs_internal.dm

```diff
@@ -67,384 +62,421 @@
+			user.visible_message("<span class='warning'> [user]'s hand slips, damaging [target]'s [affected.name] with \the [tool]!</span>", \
+			"<span class='warning'> Your hand slips, damaging [target]'s [affected.name] with \the [tool]!</span>")
+			affected.createwound(BRUISE, 20)
+		else
+			user.visible_message("[user] can't seem to extract anything from [target]'s [parse_zone(target_zone)]!",
+				"<span class='notice'>You can't extract anything from [target]'s [parse_zone(target_zone)]!</span>")
+		return 0
+	else if(current_type == "finish")
+		if(affected.encased)
+			var/msg = "<span class='warning'> [user]'s hand slips, bending [target]'s [affected.encased] the wrong way!</span>"
+			var/self_msg = "<span class='warning'> Your hand slips, bending [target]'s [affected.encased] the wrong way!</span>"
+			user.visible_message(msg, self_msg)
+			affected.fracture()
+		else
+			var/msg = "<span class='warning'> [user]'s hand slips, tearing the skin!</span>"
+			var/self_msg = "<span class='warning'> Your hand slips, tearing skin!</span>"
+			user.visible_message(msg, self_msg)
+		affected.createwound(BRUISE, 20)
+		return 0
 
-	end_step(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool)
-		user.visible_message("\blue [user] has separated [target]'s [target.op_stage.current_organ] with \the [tool]." , \
-		"\blue You have separated [target]'s [target.op_stage.current_organ] with \the [tool].")
 
-		var/obj/item/organ/I = target.op_stage.organ_ref
-		if(I && istype(I))
-			spread_germs_to_organ(I, user)
-			I.status |= ORGAN_CUT_AWAY
+	return 0
 
-	fail_step(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool)
-		var/obj/item/organ/external/affected = target.get_organ(target_zone)
-		user.visible_message("\red [user]'s hand slips, slicing an artery inside [target]'s [affected.name] with \the [tool]!", \
-		"\red Your hand slips, slicing an artery inside [target]'s [affected.name] with \the [tool]!")
-		affected.createwound(CUT, rand(30,50), 1)
 
-/datum/surgery_step/internal/remove_organ
+//////////////////////////////////////////////////////////////////
+//						SPESHUL AYLIUM STUPS					//
+//////////////////////////////////////////////////////////////////
 
+/datum/surgery_step/saw_carapace
+	name = "saw carapace"
 	allowed_tools = list(
-	/obj/item/weapon/hemostat = 100,	\
-	/obj/item/weapon/wirecutters = 75,	\
-	/obj/item/weapon/kitchen/utensil/fork = 20
+	/obj/item/weapon/circular_saw = 100, \
+	/obj/item/weapon/melee/energy/sword/cyborg/saw = 100, \
+	/obj/item/weapon/hatchet = 75
 	)
 
-	min_duration = 60
-	max_duration = 80
+	max_duration = 70
 
-	can_use(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool)
 
-		if (!..())
-			return 0
-		var/obj/item/organ/external/affected = target.get_organ(target_zone)
+/datum/surgery_step/saw_carapace/begin_step(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool,datum/surgery/surgery)
```

Comment:
I'm pretty sure just passing it in will give you a runtime and bring the proc to a halt